### PR TITLE
Feature/add action for retrieving chartdata

### PR DESCRIPTION
**Pull request by devhackerone**

### CODE REVIEW

File: frontend/src/store/modules/visitors/actions.js [hunk 1]
```diff
@@ -30,6 +31,29 @@ export default {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [GET_LINE_CHART_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+
+        context.commit(SET_LINE_CHART_DATA_FETCHING);
+        context.commit(GET_SELECTED_PERIOD)
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: commit doesn't return promise, also mutation shouldn't return data, use state from context object

in general all chain, that implemented below, won't work

File: frontend/src/store/modules/visitors/actions.js [hunk 2]
```diff
@@ -30,6 +31,29 @@ export default {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [GET_LINE_CHART_DATA]: (context, data) => {
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: don't call action as getter, unless it returns anything, in this particular case - it doesn't and shouldn't return anything

File: frontend/src/api/visitors/averageSessionsService.js [hunk 3]
```diff
@@ -60,3 +60,5 @@ export const averageSessionService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default averageSessionService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/averageSessionsService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/api/visitors/bounceRateService.js [hunk 4]
```diff
@@ -60,3 +60,5 @@ export const bounceRateService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default bounceRateService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/bounceRateService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/api/visitors/newVisitorsService.js [hunk 5]
```diff
@@ -60,3 +60,5 @@ export const newVisitorsService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default newVisitorsService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/newVisitorsService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/api/visitors/sessionsService.js [hunk 6]
```diff
@@ -60,3 +60,5 @@ export const sessionsService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default sessionsService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/sessionsService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/api/visitors/totalVisitorsService.js [hunk 7]
```diff
@@ -60,3 +60,5 @@ export const totalVisitorsService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default totalVisitorsService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/totalVisitorsService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/api/visitors/visitsService.js [hunk 8]
```diff
@@ -60,3 +60,5 @@ export const visitsService = {
     fetchChartValues,
     fetchTableValues
 };
+
+export default visitsService;
```
Comment (TatsuUkraine) [inline frontend/src/api/visitors/visitsService.js]: if you going to export this kind of default, you no need to export const above

File: frontend/src/pages/Visitors.vue [hunk 9]
```diff
@@ -71,7 +71,7 @@
     import {mapGetters} from 'vuex';
     import {
         GET_PIE_CHART_DATA,
-        GET_LINE_CHART_DATA,
+        FETCH_LINE_CHART_DATA,
```
Comment (TatsuUkraine) [inline frontend/src/pages/Visitors.vue]: getter should fetch anything

File: frontend/src/store/modules/visitors/types/getters.js [hunk 10]
```diff
@@ -5,5 +5,7 @@ export const GET_TABLE_DATA_FETCHING = 'getTableDataFetching';
 export const GET_BUTTON_DATA = 'getButtonData';
 export const GET_ACTIVE_BUTTON = 'getActiveButton';
 export const GET_PIE_CHART_DATA = 'getPieChartData';
-export const GET_LINE_CHART_DATA = 'getChartData';
-export const FETCH_TABLE_DATA = 'getTableData';
+export const GET_LINE_CHART_ITEMS = 'getLineChartItems';
+export const GET_LINE_CHART_FETCHING = 'getLineChartFetching';
+export const FETCH_LINE_CHART_DATA = 'getChartData';
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/types/getters.js]: getters will never fetch anything so getter  name, that starts with `FETCH_` doesn't make sense

File: frontend/src/api/visitors/averageSessionsService.js (hunk 3 above)
Comment (devhackerone) [reply frontend/src/api/visitors/averageSessionsService.js]: removed all export default ...Service;

File: frontend/src/pages/Visitors.vue (hunk 9 above)
Comment (devhackerone) [reply frontend/src/pages/Visitors.vue]: back GET_LINE_CHART_DATA

File: frontend/src/store/modules/visitors/actions.js [hunk 11]
```diff
@@ -40,6 +46,34 @@ export default {
...
+            context.commit(RESET_LINE_CHART_FETCHING);
+        }
+    },
+    [FETCH_LINE_CHART_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+        context.commit(SET_LINE_CHART_FETCHING);
+
+        getTimeByPeriod(context.state.selectedPeriod)
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: again, `getTimeByPeriod` doesn't return Promise, it returns object. So it won't have `then` method, so this action won't work

File: frontend/src/store/modules/visitors/actions.js [hunk 12]
```diff
@@ -57,7 +91,7 @@ export default {
         }
         context.commit(SET_TABLE_FETCHING);
 
-        getTimeByPeriod(context.state.selectedPeriod)
+        return getTimeByPeriod(context.state.selectedPeriod)
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: again, `getTimeByPeriod` doesn't return Promise, it returns object. So it won't have `then` method, so this action won't work

File: frontend/src/store/modules/visitors/actions.js [hunk 13]
```diff
@@ -40,6 +46,36 @@ export default {
...
+    },
+    [FETCH_LINE_CHART_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+        context.commit(SET_LINE_CHART_FETCHING);
+        const period = getTimeByPeriod(context.state.selectedPeriod);
+        const startDate = period.startDate;
+        const endDate = period.endDate;
+        let chartData = [];
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: this variable is redundant

File: frontend/src/store/modules/visitors/actions.js [hunk 14]
```diff
@@ -40,6 +46,36 @@ export default {
...
+
+        return factoryVisitorsService.create(context.state.activeButton)
+            .fetchChartValues(startDate.unix(), endDate.unix(), data.groupedParameter)
+                .then(response => {
+                    chartData.items = response.data;
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: just for future - Array and Object it's to different stuff, if you want to set attributes in variable, don't define it as array

**Pull request merged**